snuba/migrations/snuba_migrations/system/0001_migrations.py

```diff
@@ -13,8 +13,9 @@ def forwards_local(self) -> Sequence[operations.Operation]:
                 storage_set=StorageSetKey.MIGRATIONS,
                 statement="""\
                     CREATE TABLE migrations_local (group String, migration_id String,
-                    timestamp DateTime, status Enum('completed' = 0, 'in_progress' = 1))
-                    ENGINE = ReplacingMergeTree(timestamp) ORDER BY (group, migration_id);
+                    timestamp DateTime, status Enum('completed' = 0, 'in_progress' = 1,
+                    'not_started' = 2)) ENGINE = ReplacingMergeTree(timestamp) ORDER
```

Comment:
I think there is an issue here. Not with the enum but with the version. The granularity of this timestamp is seconds. It is possible (actually likely) to run into two steps of the same migration in the same second, then you do not know what clickhouse will do.
I would advise to replace this (now that none is using it yet) with a logical version. 
I would have used the status but it is not monotonic. So I would advise for keeping a counter when you execute multiple steps in the migration and increment it. If you restart later you would still have to load the migration status to begin with so you know where to restart from.

Something like 
```
set_not_started -> status = 0 
set_inn progress -> status = 1
BOOM
restart
load table -> status = 1
revert_to_not_started -> status = 2
set_in_progress -> status = 3
set_complete -> status = 4
```

Reply:
What do you think of using DateTime64 instead of implementing our own logic for versioning? It's new in 20.1 though so it would require an upgrade.

Reply:
It still technically not 100% correct. Since adding a logical clock is not really more complex I would still err on the safe side. Then correctness will be  easier to assess.